webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/internal/NodeDetailsAPI.scala

```diff
@@ -0,0 +1,149 @@
+package com.normation.rudder.rest.internal
+
+import com.normation.inventory.domain.NodeId
+import com.normation.inventory.domain.Software
+import com.normation.inventory.services.core.ReadOnlySoftwareDAO
+import com.normation.rudder.domain.nodes.NodeInfo
+import com.normation.rudder.domain.policies.GlobalPolicyMode
+import com.normation.rudder.domain.policies.PolicyModeOverrides.Always
+import com.normation.rudder.domain.policies.PolicyModeOverrides.Unoverridable
+import com.normation.rudder.reports.execution.AgentRunWithNodeConfig
+import com.normation.rudder.reports.execution.RoReportsExecutionRepository
+import com.normation.rudder.repository.json.DataExtractor.OptionnalJson
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.rudder.web.components.DateFormaterService
+import com.typesafe.config.ConfigRenderOptions
+import net.liftweb.common.Box
+import net.liftweb.common.Full
+import net.liftweb.common.Loggable
+import net.liftweb.http.JsonResponse
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import net.liftweb.http.rest.RestHelper
+import net.liftweb.json.JsonAST.JArray
+import net.liftweb.json.JsonAST.JField
+import net.liftweb.json.JsonAST.JNothing
+import net.liftweb.json.JsonAST.JObject
+import net.liftweb.json.JsonAST.JString
+import net.liftweb.json.parse
+import zio.ZIO
+
+class NodeDetailsAPI (
+  nodeInfoService: NodeInfoService
+  , restExtractor : RestExtractorService
+  , reportsExecutionRepository: RoReportsExecutionRepository
+  , getGlobalMode : () => Box[GlobalPolicyMode]
+  , readOnlySoftwareDAO: ReadOnlySoftwareDAO
+
+) extends  RestHelper with  Loggable {
```

Comment:
Why not use our common rest framework? 

Why not use existing node details API ? (with a new parameter to limit to list of nodes ?) Is it too much different ?

Reply:
I thought i would need a different api for the table, to me it's a private API that is optimised for the UI, and also it can be updated to integrated server-side-pagination and i think even more it will not be ok with the public API

Reply:
even with all that, it should be in `EndpointDefinition` with the scope `internal`. It's their for that case exactly. But with that, we have only one place to look for all API, internal or public (or both), and it's much easier to be sure of consistantly apply transversal needs (security, auto doc, etc). Plus: if we change from lift to something else, there's just that one place to change.

Reply:
and in the case where the API is stable enought/interestng enought, it's easier to make it public (just change the scope)